test/src/generators/app.test.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as path from 'path';
+import * as helpers from 'yeoman-test';
+import * as shell from 'shelljs';
+import { expect } from 'chai';
+
+const isWindows = process.platform === 'win32';
+const rmrf = isWindows ? 'rimraf' : 'rm -rf';
+const rmf = isWindows ? 'rimraf' : 'rm -f';
+
+const expectedPackageJSON = {
+  name: 'sfdx-plugin-org',
+  description: 'An sfdx plugin to create scratch orgs',
+  author: 'John Doe',
+  version: '0.0.1',
+  license: 'MIT',
+  files: ['/lib', '/messages', '/npm-shrinkwrap.json', '/oclif.manifest.json'],
+  homepage: 'https://github.com/johndoe/sfdx-plugin-org',
+  keywords: ['sfdx-plugin'],
+  oclif: {
+    commands: './lib/commands',
+    bin: 'sfdx',
+    topics: {
+      hello: {
+        description: 'Commands to say hello.',
+      },
+    },
+    devPlugins: ['@oclif/plugin-help'],
+  },
+  repository: 'johndoe/sfdx-plugin-org',
+  scripts: {
+    build: 'tsc -p .',
+    lint: 'eslint src/**/*.ts test/**/*.ts',
+    postpack: `${rmf} oclif.manifest.json`,
+    posttest: 'eslint src/**/*.ts test/**/*.ts',
+    prepack: `${rmrf} lib && tsc -b && oclif-dev manifest && oclif-dev readme`,
+    test: 'nyc --extension .ts --require ts-node/register mocha --forbid-only "test/**/*.test.ts"',
+    version: 'oclif-dev readme && git add README.md',
+  },
+};
+
+describe('sfdx:plugin generator', () => {
+  let runResult: helpers.RunResult;
+
+  after(async () => {
+    shell.rm('-rf', runResult.cwd);
+  });
+
+  it('generates an sfdx plugin', async () => {
+    const promptAnswers = {
+      name: 'sfdx-plugin-org',
+      description: 'An sfdx plugin to create scratch orgs',
+      author: 'John Doe',
+      version: '0.0.1',
+      license: 'MIT',
+      'github.user': 'johndoe',
+      'github.repo': 'sfdx-plugin-org',
+    };
+    runResult = await helpers
+      .run(path.join(__dirname, '../../../src/generators/app.ts'))
+      .withOptions({ skipInstall: true })
```

Comment:
Don't run install dependencies for unit tests (we already test that in the NUT!) to speed up testing

There are a few places that mention that `skip-install` is actually included in yeoman:
code: https://github.com/yeoman/generator/search?q=skip-install
docs: https://yeoman.io/authoring/integrating-yeoman.html

anyway, I didn't find a way to make it work so I added some code here:
https://github.com/forcedotcom/sfdx-plugin-generate/blob/b9b78243c8de65191316902656ed413311a11c1d/src/generators/app.ts#L330-L333
to do an early return if we pass this option (defaults to false, we only set it in this unit test).